### PR TITLE
REGR: appending to existing excel file created corrupt files

### LINKED ISSUE #39576: BUG: ExcelWriter with mode='a' corrupts file
- [x] I have checked that this issue has not already been reported.

- [x]  I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
import pandas as pd

df = pd.DataFrame()

with pd.ExcelWriter('test.xlsx', mode='w', engine='openpyxl') as writer:
    df.to_excel(writer)

with pd.ExcelWriter('test.xlsx', mode='a', engine='openpyxl') as writer:
    df.to_excel(writer)

```

#### Problem description

Appending to excel file with pandas>=1.2 produces file which Excel reports as corrupted and opens only after repair, forcing you to first open and repair file before passing it to someone else.
Looking at file with 7-Zip File Manager shows that after writing file with mode='a', some .xml files (app.xml, core.xml, theme1.xml, sheet1.xml) inside archive are duplicated (somehow). After Excel rapars file duplicates are gone, everything else looks the same.

With pandas 1.1.5 and same versions of other packages everything works as expected.

#### Expected Output

File which could be opened by Excel without any warnings.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 9d598a5e1eee26df95b3910e3f2934890d062caa
python           : 3.8.6.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19041
machine          : AMD64
processor        : Intel64 Family 6 Model 158 Stepping 10, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : Russian_Russia.1251
pandas           : 1.2.1
numpy            : 1.20.0
pytz             : 2021.1
dateutil         : 2.8.1
pip              : 21.0.1
setuptools       : 49.6.0.post20210108
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.6.2
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : 3.0.6
pandas_gbq       : None
pyarrow          : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by twoertwein**
- [x] closes #39576
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

Tentatively ready: I added a working test but I don't have MS Excel to make sure it is happy now. Waiting for someone to test it https://github.com/pandas-dev/pandas/issues/39576#issuecomment-774102850

### CODE REVIEW

File: pandas/io/excel/_openpyxl.py [hunk 1]
```diff
@@ -40,6 +40,8 @@ def __init__(
             from openpyxl import load_workbook
 
             self.book = load_workbook(self.handles.handle)
+            self.handles.handle.seek(0)
+            self.handles.handle.truncate()
```
Comment (twoertwein) [inline pandas/io/excel/_openpyxl.py:44]: that could be a bad idea if later some error happens (user's data is lost). Is there a guarantee that the appended file is always larger then the existing file? Intuitively, that is probaly the case most of the times but there might be cases where that is not correct?

Comment (twoertwein) [reply pandas/io/excel/_openpyxl.py:44]: might be better to call `self.handles.handle.truncate(self.handles.handle.tell())` in `def close()`.

**Comment by jreback** [comment]
cc @simonjayhawkins ok to merge on green

**Comment by jreback** [comment]
thanks @twoertwein

**Comment by jreback** [comment]
@meeseeksdev backport 1.2.x

**Comment by jreback** [comment]
cc @twoertwein or @simonjayhawkins for backport

**Pull request merged**